app/routes.py

```diff
@@ -47,9 +47,11 @@ def search(moods, genres):
 @app.route('/book/<book_id>')
 def show_book(book_id):
     book = Book.query.get(int(book_id))
+    check_bookmark = Bookmark.query.filter_by(book_id=book_id).first()

```

Comment:
You only want to show it as _bookmarked_, if the current user has bookmarked it.

```
if not current_user.is_anonymous():
    check_bookmark  = Bookmark.query.filter_by(book_id=book_id, user_id=current_user.id).first()
```

Reply:
Also put this after the 

```
     if book is None:
         abort(404)
```

check, otherwise book.id might my None